### PR TITLE
OSD-23925: flag customer data facing scripts

### PR DESCRIPTION
**Pull request by typeid**
### What type of PR is this?

feature

### What this PR does / Why we need it?

This PR flags user data accessing scripts as such as part of the lockbox enablement. The flag implementation will be done as part of https://issues.redhat.com/browse/OSD-22636 

### Which Jira/Github issue(s) does this PR fix?

Resolves https://issues.redhat.com/browse/OSD-23925

### Pre-checks (if applicable)

- [ ] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

**Comment by RaphaelBut** [comment]
Nice work @typeid , I think we will need to add the new label to the schema validation with something like 
```
+    "customerDataAccess": {
+      "type": "boolean",
+      "description": "Wether or not this script has access to customer data"
     }
```
and make it a required field. Wdyt?

**Comment by typeid** [comment]
/hold
Waiting on https://issues.redhat.com/browse/OSD-22636

File: hack/metadata.schema.json [hunk 1]
```diff
@@ -112,7 +112,11 @@
...
       }
-    }
+    },
+    "customerDataAccess": {
+      "type": "boolean",
+      "description": "Wether or not this script has access to customer data"
+  }
   },
   "additionalProperties": false,
   "required": [
```
Comment (RaphaelBut) [inline hack/metadata.schema.json:122]: I would say lets make it a required field to have it consistently in every script. We could have it optionally and only fail when its missing and the target cluster is access protected. But I would be strongly in favor of making it required, so its easier to build on top of it later. wdyt?

**Pull request merged**